### ISSUE TITLE
success callback called when trying to connect to unreachable host-port

### ISSUE THREAD

**Issue by christomitov** [Tlantic/cdv-socket-plugin#61]
On Mac, I enable firewall and block a certain port on the machine. From the device side I get a successCallback using the demo code you have when trying to connect to the machine. From another machine I netcat to the same host and port and as expected get a TIMEDOUT. When I disable the firewall, everything works as expected.

**Comment by felipekm** [comment]
HI @christomitov, you need some help on this?
Which platform are you this for?

Thanks for your input on this.

**Comment by christomitov** [comment]
This is both iOS and Android. You can actually try by just connecting to a random port on a random host  through the inspector and it will return success. Let me know if you need any more details.

**Comment by felipekm** [comment]
Yup sure, I will look through this, once I got some time.
Feel free to raise PR.

Thanks,

**Comment by christomitov** [comment]
Also, I don't understand objective-c very well but I suspect: https://github.com/Tlantic/cdv-socket-plugin/blob/master/src/ios/Connection.m#L29  , you need to handle some errors when reading this stream I guess? as per https://developer.apple.com/Library/ios/documentation/Networking/Conceptual/CFNetwork/CFStreamTasks/CFStreamTasks.html

**Comment by felipekm** [comment]
You got this only on iOS?

**Comment by christomitov** [comment]
No I got it in Android as well, I'm just showing specifically to the iOS platform what I suspect the issue is. I can only assume the implementation is similar on the Android side of things around here https://github.com/Tlantic/cdv-socket-plugin/blob/master/src/android/Connection.java#L114

Also 136-140 doesn't trigger any cleanup.

**Comment by christomitov** [comment]
Okay so I've circumvented this by sending a dummy "send" signal to check connection and if the host is reachable at that port. I realized from the Apple docs that opening a socket connection doesn't actually resolve the hostname, or make the connection, until the actual stream is read or written to. I assumed that the behaviour would be like netcat where when you do "nc ip port" it times out if either the host or port is unreachable. My bad!
